### PR TITLE
Fix: Smart cropping for existing images

### LINKED ISSUE #213: Smart Cropping Existing Images in Admin
Currently users have the ability to rescan images with Describe APIs via the 'Azure Computer Vision Scan' metabox within attachments. 

Could we add an additional checkbox to 'Regenerate Thumbnails' with generateThumbnail API?

https://rickal.ee/eDu1XxeJ

### PR DESCRIPTION
**Pull request by ShahAaron**
### Description of the Change

This PR adds the ability to Smart Crop existing images in WP Admin. It relates to https://github.com/10up/classifai/issues/213.

### Scope/Approach
- [x] Adds a button to the attachment page that will initiate the regeneration of thumbnails.

### Benefits

Allows Smart cropping of pre-existing images from the media attachments screen.

### Verification Process

- Under _ClassifAI -> Image Processing_ make sure _enable smart cropping_ is unchecked.
- Add an image to the media gallery
- Go back to _ClassifAI -> Image Processing_ and set _enable smart cropping_ to checked.
- Open the existing image in the media attachments screen
- Press the button to **ClassifAI Smart Crop**
- New smart cropped image attachment should be generated

**NOTE: I could not check the last step as this needs to be done on a remote server. I was able to track the functionality through the code.**

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### Applicable Issues

https://github.com/10up/classifai/issues/213

### CODE REVIEW

**Comment by ShahAaron** [comment]
@dinhtungdu I cannot check the smart cropping as I needs to be on a remote server. Please advise on how you would like me to proceed.

File: includes/Classifai/Services/ImageProcessing.php [hunk 1]
```diff
@@ -48,11 +48,13 @@ public function enqueue_media_scripts() {
...
 	 */
 	public function add_rescan_button_to_media_modal( $form_fields, $post ) {
-		$screen = get_current_screen();
+		$screen   = get_current_screen();
+		$settings = get_option( 'classifai_computer_vision' );
 		// Screen returns null on the Media library page.
 		if ( ! $screen ) {
 			$alt_tags_text   = empty( get_post_meta( $post->ID, '_wp_attachment_image_alt', true ) ) ? __( 'Generate', 'classifai' ) : __( 'Rescan', 'classifai' );
 			$image_tags_text = empty( wp_get_object_terms( $post->ID, 'classifai-image-tags' ) ) ? __( 'Generate', 'classifai' ) : __( 'Rescan', 'classifai' );
+			$smart_crop_text = empty( get_transient( 'classifai_azure_computer_vision_smart_cropping_latest_response' ) ) ? __( 'Generate', 'classifai' ) : __( 'Rescan', 'classifai' );
```
Comment (dinhtungdu) [inline includes/Classifai/Services/ImageProcessing.php]:
```suggestion
			$smart_crop_text = empty( get_transient( 'classifai_azure_computer_vision_smart_cropping_latest_response' ) ) ? __( 'Generate', 'classifai' ) : __( 'Regenerate', 'classifai' );
```

**Review by dinhtungdu — CHANGES_REQUESTED** [review]
This PR is working great on my live server. I have a concern about wording only, otherwise this is ready to be merged.

**Comment by ShahAaron** [comment]
@dinhtungdu I've just noticed I've added this to the media modal:

Screenshot 2020-10-28 at 14 21 47

But not to the media edit page:

Screenshot 2020-10-28 at 14 18 17

I'll get this done also

File: includes/Classifai/Services/ImageProcessing.php (hunk 1 above)
Comment (ShahAaron) [reply includes/Classifai/Services/ImageProcessing.php]: @dinhtungdu I've updated the above wording

**Comment by ShahAaron** [comment]
@dinhtungdu I've also added the functionality to the media edit page

File: includes/Classifai/Providers/Azure/ComputerVision.php [hunk 2]
```diff
@@ -88,8 +88,10 @@ public function setup_attachment_meta_box() {
 	 * @param \WP_Post $post The post object.
 	 */
 	public function attachment_data_meta_box( $post ) {
-		$captions = get_post_meta( $post->ID, '_wp_attachment_image_alt', true ) ? __( 'Rescan Alt Text', 'classifai' ) : __( 'Scan Alt Text', 'classifai' );
-		$tags     = ! empty( wp_get_object_terms( $post->ID, 'classifai-image-tags' ) ) ? __( 'Rescan Tags', 'classifai' ) : __( 'Generate Tags', 'classifai' );
+		$settings   = get_option( 'classifai_computer_vision' );
+		$captions   = get_post_meta( $post->ID, '_wp_attachment_image_alt', true ) ? __( 'Rescan Alt Text', 'classifai' ) : __( 'Scan Alt Text', 'classifai' );
+		$tags       = ! empty( wp_get_object_terms( $post->ID, 'classifai-image-tags' ) ) ? __( 'Rescan Tags', 'classifai' ) : __( 'Generate Tags', 'classifai' );
+		$smart_crop = get_transient( 'classifai_azure_computer_vision_smart_cropping_latest_response' ) ? __( 'Regenerate Smart Crop', 'classifai' ) : __( 'Generate Smart Crop', 'classifai' );
```
Comment (dinhtungdu) [inline includes/Classifai/Providers/Azure/ComputerVision.php]: another wording comment for this, "Smart Crop" is not straightforward for end-users. May be "Generate Smart Thumbnail" or just  "Generate Thumbnail"?

Comment (ShahAaron) [reply includes/Classifai/Providers/Azure/ComputerVision.php]: @dinhtungdu changed as requested to "Generate Smart Thumbnail"

**Review by dinhtungdu — APPROVED** [review]
LGTM 🎉

**Pull request merged**